src/dune_2_excel.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+from typing import List, Any
+from datetime import datetime
+
+import click
+from dotenv import load_dotenv
+from dune_client.client import DuneClient
+from dune_client.models import ResultsResponse
+from dune_client.query import Query
+from dune_client.types import QueryParameter
+import pandas as pd
+from tqdm import tqdm
+
+
+@click.command()
+@click.option("--query", "-q", multiple=True)
+@click.option("--start-date", "-s", type=click.DateTime(formats=["%Y-%m-%d"]))
+def main(query: List[str], start_date: datetime) -> Any:
+    """
+    Main function of the script
+    Args:
+        query: List of queries to fetch
+        start_date: start_date query parameter
+
+    Returns:
+
+    """
+    monthly_reporting(query, start_date)
+    return 0
+
+
+def store_results(results: List[ResultsResponse], start_date: datetime) -> None:
+    """
+    Store results into xlsx file
+    Args:
+        results: results to be stored
+        start_date: start_date query parameter
+    """
+    writer = pd.ExcelWriter(  # pylint: disable=abstract-class-instantiated
+        f'out/{"_".join([str(x.query_id) for x in results])}'
+        f'_{start_date.strftime("%Y-%m-%d")}.xlsx',
+        engine="xlsxwriter",
+    )
+    for result in results:
+        pd.DataFrame(pd.DataFrame(result.result.rows)).to_excel(
+            writer, sheet_name=f"{result.query_id}", index=False
```

Comment:
Would love to add query names as sheet names instead of query id's here! But I can't seem to find how you did it in https://github.com/cowprotocol/cow-dune/pull/60 @bh2smith

Reply:
I see you are using dune-client, there is a convenience method available so you don't have to access them as `result.result.rows` (should be something like `result.get_rows()` - [here](https://github.com/cowprotocol/dune-client/pull/26/files#r1012305477)

I don't think you have access to the query name since the input only provides query ID. The names are set when Query object is defined (for logging purposes). The legacy API can be used to fetch query metadata.

Reply:
> I see you are using dune-client, there is a convenience method available so you don't have to access them as result.result.rows (should be something like result.get_rows() - [here](https://github.com/cowprotocol/dune-client/pull/26/files#r1012305477)

Thanks added your suggestions.

> I don't think you have access to the query name since the input only provides query ID. The names are set when Query object is defined (for logging purposes). The legacy API can be used to fetch query metadata.

Ah okay. Let's leave it for this like now.